### PR TITLE
Rewrite the docs

### PR DESCRIPTION
**Pull request by Peregrine05**
Rewrote the docs.

- Reorganized the articles, and updated and corrected the information contained therein.

### CODE REVIEW

**Comment by jackjt8** [comment]
My review will take a while given the scope of the changes however I have been updated over the course of these changes being made so it might not take forever.

It may be worth requesting a review from some of the other maintainers due to the scope of the changes.

**Comment by Peregrine05** [comment]
I have requested reviews from leMaik and Redox, since they have also worked on the docs, and have a good understanding of how Chunky, especially the more technical aspects of it, function.

File: ChunkyDocs/docs/reference/technical/scene_format.md
Comment (jackjt8) [inline ChunkyDocs/docs/reference/technical/scene_format.md:85]: TODO - Update contents of SDF. ie `emitterSamplingStrategy | {“NONE”, “ONE”, “ALL”}` for 2.5.0.

Please defer this to a new issue as this won't block merging of this PR but this does need to be updated at some point.

File: ChunkyDocs/docs/plugins/plugin_list.md
Comment (jackjt8) [inline ChunkyDocs/docs/plugins/plugin_list.md:57]: "GPU support is on hold for now" - Redox

Adjust wording to reflect that while there is a plugin it is not in active development.

File: ChunkyDocs/docs/faq/index.md
Comment (jackjt8) [inline ChunkyDocs/docs/faq/index.md:21]: "GPU support is on hold for now" - Redox

Adjust wording to reflect that while there is a plugin it is not in active development.

File: README.md
Comment (Peregrine05) [reply README.md:18]: I must have copy-pasted from the `gallery` repo without changing it. It should be `ChunkyDocs`.

File: ChunkyDocs/docs/getting_started/configuring_chunky_launcher.md
Comment (Peregrine05) [reply ChunkyDocs/docs/getting_started/configuring_chunky_launcher.md:29]: No, the correct sentence is "It is recommended that this remain enabled."

File: ChunkyDocs/docs/reference/user_interface/render_controls/materials.md
Comment (Peregrine05) [reply ChunkyDocs/docs/reference/user_interface/render_controls/materials.md:43]: The image is not composite. The blocks that make up the side walls simply have different material properties in the same render.

File: ChunkyDocs/docs/reference/user_interface/render_controls/materials.md
Comment (Peregrine05) [reply ChunkyDocs/docs/reference/user_interface/render_controls/materials.md:54]: The image is not composite. The blocks that make up the side walls simply have different material properties in the same render.

File: ChunkyDocs/docs/reference/user_interface/render_controls/materials.md
Comment (Peregrine05) [reply ChunkyDocs/docs/reference/user_interface/render_controls/materials.md:87]: The image is not composite. The blocks that make up the side walls simply have different material properties in the same render.

File: ChunkyDocs/docs/reference/introduction/next_event_estimation.md
Comment (Peregrine05) [reply ChunkyDocs/docs/reference/introduction/next_event_estimation.md:86]: I changed it to ". . . and `ALL`. With `ESS: NONE`, ESS is disabled," instead.

File: ChunkyDocs/docs/reference/introduction/material_properties.md
Comment (Peregrine05) [reply ChunkyDocs/docs/reference/introduction/material_properties.md:5]: A comma should follow "In Chunky". No comma should should follow "For every material".

File: ChunkyDocs/docs/reference/introduction/material_properties.md
Comment (Peregrine05) [reply ChunkyDocs/docs/reference/introduction/material_properties.md:11]: I have changed it to "Light emitted from materials which have an emittance value that is greater than *0* is colored, overall, according to the average of all color values of the texture of that material."

File: ChunkyDocs/docs/reference/user_interface/map.md
Comment (Peregrine05) [reply ChunkyDocs/docs/reference/user_interface/map.md:190]: The documentation for the other controls is in the `Map View` tab documentation. I only document controls that have changed in some way in the 2.5.0 Snapshot sections. The `2.5.0 Snapshot Controls` section of the `Map View` tab links to the section in the `Map` tab to further document any changes.

File: ChunkyDocs/docs/reference/user_interface/render_controls/scene.md
Comment (Peregrine05) [reply ChunkyDocs/docs/reference/user_interface/render_controls/scene.md:79]: This control is not labeled in the GUI, so I must use an image to represent it, as in `Scene Management`.

File: ChunkyDocs/docs/reference/user_interface/render_controls/scene.md
Comment (Peregrine05) [reply ChunkyDocs/docs/reference/user_interface/render_controls/scene.md:121]: Trimmed and aligned with `vertical-align: middle;`

File: ChunkyDocs/docs/reference/user_interface/render_controls/sky_and_fog.md
Comment (Peregrine05) [reply ChunkyDocs/docs/reference/user_interface/render_controls/sky_and_fog.md:113]: I separated the figures in this instance to keep the structure of the `Sky mode settings` section intact. The figures are linked to from the sections to which they pertain.

File: ChunkyDocs/docs/reference/user_interface/render_controls/sky_and_fog.md
Comment (Peregrine05) [reply ChunkyDocs/docs/reference/user_interface/render_controls/sky_and_fog.md:125]: They are very related, so I see no problem in doing so. It also saves space.

File: ChunkyDocs/docs/reference/user_interface/render_controls/camera.md
Comment (Peregrine05) [reply ChunkyDocs/docs/reference/user_interface/render_controls/camera.md:21]: The icons were improved in chunky-dev/chunky#1377, so they can be added.

File: ChunkyDocs/docs/reference/other_topics/denoising.md
Comment (Peregrine05) [reply ChunkyDocs/docs/reference/other_topics/denoising.md:23]: True. I'll remove it.

File: ChunkyDocs/docs/reference/other_topics/denoising.md
Comment (Peregrine05) [reply ChunkyDocs/docs/reference/other_topics/denoising.md:25]: True. I'll remove it.

File: ChunkyDocs/docs/reference/other_topics/styling_chunky.md
Comment (Peregrine05) [reply ChunkyDocs/docs/reference/other_topics/styling_chunky.md:93]: Custom theme: "Light", by jackjt8. *Sorry*

File: ChunkyDocs/docs/reference/other_topics/styling_chunky.md
Comment (Peregrine05) [reply ChunkyDocs/docs/reference/other_topics/styling_chunky.md:175]: Custom theme: "Nice Blue", by EmeraldSnorlax. *MIA*

File: ChunkyDocs/docs/reference/technical/scene_format.md
Comment (Peregrine05) [reply ChunkyDocs/docs/reference/technical/scene_format.md:7]: They have now been added to the abbreviations file.

File: ChunkyDocs/docs/extra/extra.css
Comment (Peregrine05) [reply ChunkyDocs/docs/extra/extra.css:206]: The margins and padding have been reduced, as well as the size of the text.

File: ChunkyDocs/docs/reference/technical/scene_format.md
Comment (Peregrine05) [reply ChunkyDocs/docs/reference/technical/scene_format.md:85]: An issue has been created.

File: ChunkyDocs/docs/reference/introduction/material_properties.md
Comment (jackjt8) [reply ChunkyDocs/docs/reference/introduction/material_properties.md:5]: My English teacher would have shot me for this.

"a punctuation mark that separates words, clauses, or ideas within a sentence"

When reading the first sentence aloud the soft-break makes it unnatured. Either we drop the comma or we drop the In Chunky part.


`For every material these properties can be set in the Materials tab.` This sentence works fine. `except for beacon beam segments` is additional information that could be in a separate sentence.  Currently this sentence goes against what a comma is for.

A & B, something related to A not B. <- How does that make sense?

File: ChunkyDocs/docs/reference/user_interface/render_controls/scene.md
Comment (jackjt8) [reply ChunkyDocs/docs/reference/user_interface/render_controls/scene.md]: At the very least we need to adjust the scaling of this image. It stands out too much compared to everything else.

I'll open an issue on Chunky to try and get this labelled.

Comment (jackjt8) [reply ChunkyDocs/docs/reference/user_interface/render_controls/scene.md]: Actually maybe we just need to change the drop down to have 0 so we don't have two control methods for one action.

File: ChunkyDocs/docs/reference/introduction/material_properties.md
Comment (Peregrine05) [reply ChunkyDocs/docs/reference/introduction/material_properties.md:5]: The first sentence can be changed to, "Every block and 'entity' in Chunky is listed as a material."
The second sentence can be changed to, "Material properties for beacon beam segments can be set in the Entities tab, and material properties for every other material can be set in the Materials tab."

File: ChunkyDocs/docs/reference/user_interface/render_controls/scene.md
Comment (Peregrine05) [reply ChunkyDocs/docs/reference/user_interface/render_controls/scene.md]: I changed it to `..."X" frames` to also indicate that it continues where the previous control name left off.

**Review by jackjt8 — APPROVED** [review]
If one more person Approves these changes I will merge them.

Just because of the PR size mainly...

File: ChunkyDocs/docs/plugins/chunky_plugins.md
Comment (ThatRedox) [inline ChunkyDocs/docs/plugins/chunky_plugins.md:17]: I think we should remove this and only support adding plugins through the plugin manager.

File: ChunkyDocs/docs/plugins/plugin_development.md
Comment (ThatRedox) [inline ChunkyDocs/docs/plugins/plugin_development.md:100]: Maybe consider replacing these with `ChunkyAOV` since they use the old, deprecated interface for setting renderers?

File: ChunkyDocs/docs/getting_started/installing_chunky.md
Comment (jackjt8) [reply ChunkyDocs/docs/getting_started/installing_chunky.md:74]: To my knowledge there is no indication to the end user that this layer is used unless you dig for it.

Maybe we should relocate it to troubleshooting or something? While it is outside of the scope of Chunky itself it can still have an impact on the user and it's more food for thought in that sense.

File: ChunkyDocs/docs/getting_started/configuring_chunky_launcher.md
Comment (Peregrine05) [reply ChunkyDocs/docs/getting_started/configuring_chunky_launcher.md:113]: Changed to:

- `-spp-per-pass <NUM>`: The spp-per-pass defines the number of samples a certain tile should be rendered to before moving on to the next tile. The default value of *1* means that each pixel will be sampled once per pass. This results in the render preview displaying the most recent render progress, and responding to changes after only one pass is rendered. Raising the spp-per-pass breaks some of the GUI functionality; however, rendering performance may be improved. It is recommended that this option be only used for headless operation.

File: ChunkyDocs/docs/getting_started/configuring_chunky_launcher.md
Comment (Peregrine05) [reply ChunkyDocs/docs/getting_started/configuring_chunky_launcher.md:30]: Changed to:

- `Always open Launcher`: Changes whether the Launcher is shown when starting Chunky. If it becomes disabled, it is possible to access the launcher again via the command line or an option in Chunky. This is slightly more complicated, however, so it is recommended to keep this option enabled.

File: ChunkyDocs/docs/plugins/chunky_plugins.md
Comment (Peregrine05) [reply ChunkyDocs/docs/plugins/chunky_plugins.md:17]: Moving a plugin to the "plugins" folder is exactly what the plugin manager does, so there is no reason to remove this step, since it is a simpler method for some users to install plugins.

File: ChunkyDocs/docs/plugins/plugin_development.md
Comment (Peregrine05) [reply ChunkyDocs/docs/plugins/plugin_development.md:100]: Changed to:

To demonstrate some features of the Plugin API, llbit created a few demo plugins.

- <a href="https://github.com/llbit/Chunky-AOPlugin" target="_blank">Ambient Occlusion Plugin</a>

- <a href="https://github.com/llbit/Chunky-DepthPlugin" target="_blank">Depth Buffer Plugin</a>

- <a href="https://github.com/llbit/Chunky-BlockMod" target="_blank">Block Plugin Template</a>

- <a href="https://github.com/llbit/Chunky-TabMod" target="_blank">Tab Plugin Template</a>

The Ambient Occlusion plugin and the Depth Buffer plugin use a deprecated API to add renderers to Chunky. The Chunky AOV plugin adds these renderers using the new API.

**Pull request merged**
